cabal-install/tests/IntegrationTests2.hs

```diff
@@ -842,6 +842,17 @@ testTargetProblemsRepl config reportSubCase = do
 testTargetProblemsRun :: ProjectConfig -> (String -> IO ()) -> Assertion
 testTargetProblemsRun config reportSubCase = do
 
+    reportSubCase "one-of-each"
+    do (_,elaboratedPlan,_) <- planProject "targets/one-of-each" config
+       assertProjectDistinctTargets
+         elaboratedPlan
+         CmdRun.selectPackageTargets
+         CmdRun.selectComponentTarget
+         [ TargetPackage TargetExplicitNamed ["p-0.1"] Nothing
+         ]
+         [ ("p-0.1-inplace-p1",      CExeName   "p1")
+         ]
+
```

Comment:
Since the target selection logic is duplicated between list-bin and run it could be a good idea to have a case for list-bin too

Reply:
Done!

Sadly there's a bit duplication in the tests right now, but I think it's okay.